sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -422,7 +431,18 @@ def delete_configuration_setting(self, key, label=None, **kwargs):
                 key="MyKey", label="MyLabel"
             )
         """
+        key = kwargs.pop("key", None)
+        label = kwargs.pop("label", None)
         etag = kwargs.pop("etag", None)
+        if len(args) > 0:
+            if isinstance(args[0], ConfigurationSetting):
+                key = args[0].key
+                label = args[0].label
+                etag = args[0].etag if not None else etag
```

Comment:
if user specifies ConfigSetting and etag kwarg, which one will be preceding?

Reply:
The default is the kwarg

Reply:
So if both exist, we will use the one in ConfigSetting, right?

Reply:
If both exist we use kwarg

Reply:
https://azure.github.io/azure-sdk/python_design.html#python-method-conditional-request-etag

Reply:
Then I am afraid we need to update the code. :)